src/transaction.js

```diff
@@ -389,11 +389,65 @@ function serializeTx (txObj: TXOBJ, withPrevScriptPubKey: boolean = false): stri
     serializedTx += o.script
   })
 
-  // Only handles forward transfers at the moment
+  // Handles sidechain creation and forward transfers
   if (txObj.version == -4) {
     serializedTx += zbufferutils.numToVarInt(0); // Write every csw
-    serializedTx += zbufferutils.numToVarInt(0); // Write every vsc
-    
+
+    serializedTx += zbufferutils.numToVarInt(txObj.vsc_ccout.length); // Write every vsc
+    for (var i = 0; i < txObj.vsc_ccout.length; i++) {
+
+      const sc_c = txObj.vsc_ccout[i];
+
+      let _b = Buffer.alloc(12);
+      _b.writeUIntLE(sc_c.withdrawalEpochLength, 0, 3);
+      _b.writeIntBE(sc_c.version, 3, 1);
```

Comment:
You could also do `_b.writeInt8(sc_c.version, 3)` to keep consistent with the deserialization